src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -161,7 +166,10 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 		});
 	}
 
-	activate(terminal: Terminal): void { this._terminal = terminal; }
+	activate(terminal: Terminal): void {
+		this._terminal = terminal;
+		this._terminal.options.overviewRulerWidth = DecorationStyles.OverlayRulerWidth;
```

Comment:
This can move into XtermTerminal with the default options so it's set immediately